server/block/fire.go

```diff
@@ -1,12 +1,16 @@
 package block
 
+//lint:file-ignore ST1022 Exported variables in this package have compiler directives. These variables are not otherwise exposed to users.
+
 import (
 	"github.com/df-mc/dragonfly/server/block/cube"
 	"github.com/df-mc/dragonfly/server/entity"
 	"github.com/df-mc/dragonfly/server/entity/damage"
 	"github.com/df-mc/dragonfly/server/world"
+	"github.com/go-gl/mathgl/mgl64"
 	"math/rand"
 	"time"
+	_ "unsafe" // Imported for compiler directives.
```

Comment:
I would like to to refrain from using any more go:linkname compiler directives. We had a bit of a journey to try and get rid of them all a while back and I think it's best if we don't go down that rabbit hole again. Compiler directives make it very difficult to refactor things.

Do you reckon we can approach this in some other way? I understand the circular dependencies are a bit of a problem, but if we can avoid using compiler directives that would be very helpful in the future.